crates/hir-ty/src/lower_nextsolver.rs

```diff
@@ -1355,6 +1355,20 @@ pub(crate) fn generic_predicates_for_param_cycle_result(
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct GenericPredicates<'db>(Option<Arc<[Clause<'db>]>>);
 
+impl<'db, 'a> GenericPredicates<'db> {
+    pub fn instantiate(
```

Comment:
I think I prefer to wrap the field in `EarlyBinder` then use `iter_instantiated()`.